llvm/lib/CodeGen/StackMaps.cpp

```diff
@@ -592,18 +589,49 @@ void StackMaps::recordStackMapOpers(const MCSymbol &MILabel,
 
   const TargetRegisterInfo *TRI = AP.MF->getSubtarget().getRegisterInfo();
   std::set<int64_t> TrackedRegisters;
-  for (const auto *Op = MI.operands_begin(); Op != MI.operands_end(); Op++) {
-    if (Op->isReg() && !Op->isImplicit() && !Op->isUndef()) {
-      TrackedRegisters.insert(getDwarfRegNum(Op->getReg(), TRI));
+  if (MI.getOpcode() == TargetOpcode::PATCHPOINT) {
```

Comment:
We need a big-ish comment here explaining what's going on because I'll definitely forget. E.g. patchpoints look different than stackmaps due to the built in function call including arguments which we don't want to track and thus need to skip.

Reply:
Added :point_right: 43a2e5d7d31e4dde5234f7b30d669212ecd47e57